File: ruby/HelloWorld/test/greeter_spec.rb
require 'rspec'
require_relative '../src/greeter'


describe 'Greeter' do
  let(:greeter) { Greeter.new }

  it 'should greet the world' do
    expect(greeter.say_hello_to).to eq('Hello World!')
  end

  it 'should greet a person' do
    expect(greeter.say_hello_to('Christoph')).to eq('Hello Christoph!')
  end
end

File: javascript/demo-kata/test/greeter.spec.js
import {expect} from 'chai';
import {Greeter} from '../src/greeter';

describe('Greeter', () => {
    let greeter;

    beforeEach(() => greeter = new Greeter());

    it('should greet the world', () => {
        expect(greeter.sayHello()).to.equal('Hello, World!');
    });

    it('should greet the specified person', () => {
        expect(greeter.sayHello('Christoph')).to.equal('Hello, Christoph!');
    });
});